code/game/objects/items/weapons/cosmetics.dm

```diff
@@ -88,4 +88,87 @@
 					H.lip_style = null
 					H.update_body()
 	else
-		..()
\ No newline at end of file
+		..()
+
+/obj/item/weapon/hair_dye
+	name = "can of hair dye"
+	desc = "A can of sprayable hair dye. There is a dial on the top for color selection."
+	icon = 'icons/obj/items.dmi'
+	icon_state = "hair_dye"
+	flags = FPRINT
+	w_class = W_CLASS_SMALL
+	var/color_r = 255
+	var/color_g = 255
+	var/color_b = 255
+
+/obj/item/weapon/hair_dye/New()
+	..()
+	color_r = rand(0,255)
+	color_g = rand(0,255)
+	color_b = rand(0,255)
+	update_icon()
+
+/obj/item/weapon/hair_dye/update_icon()
+	overlays.len = 0
+	var/icon/dye_color = new/icon("icon" = 'icons/obj/items.dmi', "icon_state" = "dye_color_overlay")
+	dye_color.Blend(rgb(color_r, color_g, color_b), ICON_ADD)
+	overlays += dye_color
+
+/obj/item/weapon/hair_dye/attack_self(mob/user as mob)
+	var/new_color = input(user, "Choose the dye's color:", "Color Select") as color|null
+	if(new_color)
+		color_r = hex2num(copytext(new_color, 2, 4))
+		color_g = hex2num(copytext(new_color, 4, 6))
+		color_b = hex2num(copytext(new_color, 6, 8))
+	update_icon()
+
+/obj/item/weapon/hair_dye/attack(mob/M as mob, mob/user as mob)
+	if(!istype(M, /mob))
+		return
+
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if(isjusthuman(H) || isunathi(H))	//Humans and unathi are the only races with actual hair.
```

Comment:
Are you sure? I'm pretty sure there is, at the very least its used in icon updating